### ISSUE TITLE
IDE0009 incorrectly suggested in nameof expression used to class attribute constructor argument in VS 15.5

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn#23803]
**Version Used**: VS2017 15.5.2

**Steps to Reproduce**:

1. Open `Opetions`, select `Text Editor -> C# -> Code Style -> General` tab, set `Qualify property access with 'this'` option to `Prefer 'this.'`
2. Create Visual C# Console App Project
3. Copy and paste following code:
```cs
using System;

class MyAttribute : Attribute
{
    public MyAttribute(string name) { }
}

[My(nameof(Property))] // IDE0009 "Add 'this' or 'Me' qualification." is suggested here
class Program
{
    int Property { get; set; }

    static void Main() { }
}
```

**Expected Behavior**:
IDE0009 is not suggested.

**Actual Behavior**:
IDE0009 is suggested.
If I add "this" qualification, I get an compile error: CS0027 `Keyword 'this' is not available in the current context`.

**Comment by nvmkpk** [comment]
I reported a similar issue but at a different place where the error would not occur. But that was turned down saying it will be a costly fix. Now this has to be fixed. I would either suggest removing this warning entirely in context of 'nameof' or suggest adding the class name to qualify it instead (later one improves readability).

**Comment by davidwengier** [comment]
Fixing this in #26893
